src/Resolvers/Collections.ts

```diff
@@ -92,35 +92,38 @@ export class CollectionsResolver {
 
   @FieldResolver(type => [Collection])
   async relatedCollections(
+    @Arg("size", () => Int, { defaultValue: 10 }) size: number,
     @Root() collection: Collection
   ): Promise<Collection[]> {
     const isArtistBased =
       collection.query.artist_ids && collection.query.artist_ids.length > 0
 
     if (isArtistBased) {
       const artistResults = await this.repository.find({
-        where: { "query.artist_ids": { $in: collection.query.artist_ids } },
+        where: {
+          slug: { $ne: collection.slug },
+          "query.artist_ids": { $in: collection.query.artist_ids },
+        },
+        take: size,
       })
 
-      const relatedArtistCollections = reject(artistResults, {
-        id: collection.id,
-      })
-
-      if (relatedArtistCollections.length > 4) {
-        return relatedArtistCollections
+      // We only want to return relatedArtistCollections when there are more
+      // than 4. Otherwise we return category based related collections.
+      if (artistResults.length > 4) {
+        return artistResults
       }
     }
 
     const relatedCategoryResults = await this.repository.find({
       where: {
-        category: { $in: [collection.category] },
+        slug: { $ne: collection.slug },
+        category: { $in: collection.category },
```

Comment:
Looking good 🎉  

But I'm wondering about this line — the `$in` should only be there if the test is against an array, which is not the case here. I'd expect the line above to cause an error, and instead to see something simpler like

```js
category: collection.category,
```

Then again, tests are green. So either I'm wrong or the tests are 😬

Reply:
Looking closer at the tests, it looks like we [mock out the connection to the database](https://github.com/artsy/kaws/blob/83f412d2c40005691ab7985b5f9c5ccbfb4323b5/src/Resolvers/__tests__/collection_resolvers.test.ts#L8), so the tests wouldn't necessarily catch a malformed Mongo query.

If you'd prefer I'm happy to pair on this with you and confirm my suspicions 😄

Reply:
I went ahead and made the change but I'm down to pair in order to confirm.